### DISCUSSION TITLE
Jumping to Source or Destination Block Based on Control Flow Arrows

### DISCUSSION THREAD

**Discussion by utkonos** [Vector35/binaryninja-api#6150] [Q&A] [answered]
This may just be some shortcut that I just am not aware of, btw. If so, can you point me to the proper Vulcan death grip (or just one keypress) I need to use on the keyboard to do this?

If I have one control flow arrow selected, is there a way to move the View's focus to the source block or destination block with a shortcut? I don't see anything in a right-click menu that looks like what I want.

Given something like this:
<img width="1672" alt="image" src="https://github.com/user-attachments/assets/4ebe302c-03e3-4d87-95cf-1345b47cca23">

Or this:
<img width="1287" alt="image" src="https://github.com/user-attachments/assets/e1bdb803-420a-4620-9531-12fe097d2b2e">

**Comment by psifertex** [comment, marked as answer]
Enter should do what you want if I understand correctly.

**Reply by psifertex** [reply]
It will go forward if you selected by using arrows down from the exit and backward if using arrows up from entry. Likewise double clicking does the same thing based on always navigating to the opposite half that you clicked on.

**Reply by utkonos** [reply]
On macOS, Enter (numpad key) and Return, don't appear to have this behavior. But mouse double click does exactly what I want. I notice one tiny detail which is not too important unless there are lots of arrows leaving or arriving at a block: double click not only moves focus as I want, but it also deselects the arrow. If there are many, it's not easy to know which one brought me to the new block so I can get back quickly. But other than that minor detail, great.

**Reply by psifertex** [reply]
Actually, I was wrong, you can just push the down-arrow key again (or up-arrow), not enter! Sorry about that, I was just mis-remembering.

Yeah, de-selecting the arrow I agree is annoying. There might even be an open issue for it though a quick search didn't find one, I know it's come up before.

**Reply by utkonos** [reply]
Arrows work! Thanks. Better than double click due to the deselection.
